### DISCUSSION TITLE
Monogame support?

### DISCUSSION THREAD

**Discussion by breynard0** [AvaloniaUI/Avalonia#7437] [General]
I want to use Monogame to add 3D functionality in Avalonia. Is this possible? I don't think there is a package or anything for it, but is it possible to pull off a workaround? Is this a planned feature? Are there any other 3D libraries that do work with Avalonia?

**Comment by vchelaru** [comment]
Can any maintainers comment on this? This is the #1 issue that is keeping me from using Avalonia, as I want to use it to create tools for MonoGame.

**Reply by jp2masa** [reply]
There is `OpenGlControlBase` which could probably be used to integrate with MonoGame OpenGL API, but on Windows (WGL) it's not really stable.

**Comment by damian-666** [comment]
There any update on this? Looking all around, I see that there is a way to share  the surface.

 And ideally, to use the device provider service  Used in the. Craftwork wpf control., .public class MonoGameServiceProvider : then it should not have a game class. I do see a thing called Monogame.Inside It's really hard to get it into a docking system 

https://github.com/wieslawsoltes/Dock   cannot redock floating,,  tried dirkster but it has too many features.

Ideally you may want to share a back device that's the size of the  at maximum the Windows desktop or desktop of whatever you're using. in case you are undocking and then you and then you  resize. or if you have any transparency and you have floating and docking controls, but right now really if I just had dockable and undocuble Avalonia   tearable,  Redockable., views with monogame or kni  inside I'd be happy. I

**Reply by maxkatz6** [reply]
There are no plans to have built-in support for Monogame from Avalonia framework, so nothing to update here.
We do provide GPU interoperability support, making it possible to integrate third party rendering engines into Avalonia pipeline, if needed.

You can help [this project](https://github.com/AvaloniaInside/AvaloniaInside.MonoGame) to move this support ahead.
